src/main/kotlin/org/malachite/estella/services/ApplicationService.kt

```diff
@@ -51,17 +54,22 @@ class ApplicationService(
             interviewService.createInterview(offer, application)
             application
         } ?: throw NoSuchElementException()
-
     }
 
     fun setNextStageOfApplication(applicationId: Int) {
         val application = applicationRepository.findById(applicationId).get()
+
+        if (application.status != ApplicationStatus.IN_PROGRESS)
+            throw UnsupportedOperationException("Cannot change stage of resolved application!")
+
         val recruitmentProcessStages = recruitmentProcessService
             .getProcessFromStage(application.stage)
             .stages
             .sortedBy { it.id }
         val index = recruitmentProcessStages.indexOf(application.stage)
-        if (index + 1 < recruitmentProcessStages.size)
+        if (index == recruitmentProcessStages.lastIndex)
+            applicationRepository.save(application.copy(stage = recruitmentProcessStages[index + 1], status = ApplicationStatus.ACCEPTED))
+        else if (index + 1 < recruitmentProcessStages.size)
             applicationRepository.save(application.copy(stage = recruitmentProcessStages[index + 1]))
     }
```

Comment:
To tak, jak gadaliśmy - trzeba porozbijać do osobnych serwisów i coś z tym zrobić, bo zwyczajnie jest to brzydkie. A mnie na sysopach za brak SOLIDu krzyczałeś... XD

Reply:
ale jak do "osobnych serwisów"? nie jest jakaś piękna ta funkcja, ale nie pasuje to żadnego innego serwisu, mogę ewentualnie ją pogranulować żeby bardziej było widać co tam się dzieje